src/Tools/Source/RunTests/ProcessRunner.cs

```diff
@@ -0,0 +1,228 @@
+                    if (e.Data != null)
+                    {
+                        outputLines.Add(e.Data);
+                    }
+                };
+
+            process.ErrorDataReceived += (s, e) =>
+                {
+                    if (e.Data != null)
+                    {
+                        errorLines.Add(e.Data);
+                    }
+                };
+
+            process.Exited += (s, e) =>
+                {
+                    var processOutput = new ProcessOutput(process.ExitCode, outputLines, errorLines);
+                    taskCompletionSource.TrySetResult(processOutput);
+                };
+
+            var registration = cancellationToken.Register(() =>
+                {
+                    if (taskCompletionSource.TrySetCanceled())
+                    {
+                        // If the underlying process is still running, we should kill it
+                        if (!process.HasExited)
+                        {
+                            try
+                            {
+                                process.Kill();
+                            }
+                            catch (InvalidOperationException)
+                            {
+                                // Ignore, since the process is already dead
+                            }
+                        }
+                    }
+                });
+
+            return taskCompletionSource.Task;
+        }
+
+        private static ProcessStartInfo CreateProcessStartInfo(
+            string executable, string arguments,
+            string workingDirectory,
+            bool captureOutput,
+            bool displayWindow,
+            Dictionary<string, string> environmentVariables = null)
+        {
+            var processStartInfo = new ProcessStartInfo(executable, arguments);
+
+            if (!string.IsNullOrEmpty(workingDirectory))
+            {
+                processStartInfo.WorkingDirectory = workingDirectory;
+            }
+
+            if (environmentVariables != null)
+            {
+                foreach (var pair in environmentVariables)
+                {
+                    processStartInfo.EnvironmentVariables[pair.Key] = pair.Value;
+                }
+            }
+
+            if (captureOutput)
+            {
+                processStartInfo.UseShellExecute = false;
+                processStartInfo.RedirectStandardOutput = true;
+                processStartInfo.RedirectStandardError = true;
+            }
+            else
+            {
+                processStartInfo.CreateNoWindow = !displayWindow;
+                processStartInfo.UseShellExecute = displayWindow;
+            }
+
+            return processStartInfo;
+        }
+
+        public static ProcessStartInfo CreateElevatedStartInfo(string executable, string arguments, string workingDirectory, bool captureOutput, bool displayWindow)
+        {
+            var adminInfo = new ProcessStartInfo(executable, arguments);
```

Comment:
So you can't capture output of an elevated task?  Should you throw on that combination?  This seems odd.